core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -580,23 +580,25 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // PushJoinIntoTableScan must run after ReorderJoins and DetermineJoinDistributionType because they require
+        // stats but most connectors don't provide stats after join pushdown today
```

Comment:
Hive is the only one who provides stats and it doesn't support Join pushdown. The actual reason is different though. Since it's enough  to have comment in the other place, i will comment there.

```suggestion
```